django/db/models/fields/mixins.py

```diff
@@ -12,6 +12,12 @@ def get_cached_value(self, instance, default=NOT_PROVIDED):
         try:
             return instance._state.fields_cache[cache_name]
         except KeyError:
+            ancestor_link = instance._meta.get_ancestor_link(self.model)
+            if ancestor_link and ancestor_link.is_cached(instance):
```

Comment:
I don't see a failing test if `and ancestor_link.is_cached(instance)` is removed. Can a test be added to prevent a regression if that's removed?